src/Application.php

```diff
@@ -30,54 +31,40 @@ class Application
 
     private StorageGenerator $storageGenerator;
 
-    private ParametersGenerator $parametersGenerator;
+    private BlocksGenerator $blocksGenerator;
 
-    private ApiFacade $apiFacade;
+    private ParametersFactory $parametersFactory;
 
     public function __construct(string $dataDir, Config $config, LoggerInterface $logger)
     {
         $this->config = $config;
         $this->logger = $logger;
         $this->dataDir = $dataDir;
-        $this->patternFactory = new PatternFactory($this->config);
+        $this->apiFacade = new ApiFacade($this->config, $dataDir);
+        $this->patternFactory = new PatternFactory($this->config->getScdType());
         $this->pattern = $this->patternFactory->create();
         $this->mappingManager = new MappingManager($this->config, $this->pattern);
         $this->storageGenerator = new StorageGenerator($this->mappingManager);
-        $this->parametersGenerator = new ParametersGenerator();
-        $this->apiFacade = new ApiFacade($this->config, $this->pattern, $this->mappingManager, $dataDir);
+        $this->blocksGenerator = new BlocksGenerator();
+        $this->parametersFactory = new ParametersFactory($this->config, $this->apiFacade, $this->mappingManager);
     }
 
     public function generateConfig(): array
     {
-        $this->apiFacade->createSnapshotTable();
+        // Create parameters
+        $parameters = $this->parametersFactory->create();
+        $this->pattern->setParameters($parameters);
```

Comment:
Pridanim validacie vstupov tam zostala este jedna kruhova zavislost,  a preto nemozem vlozit Parameters do Pattern cez konstruktor ale nastavuju sa dodatocne -> cez setter ... nie je to uplne ciste ... ale inak by to boli stovky riadkov uprav.